### ISSUE TITLE
Require mandatory return_dtype on apply

### ISSUE THREAD

**Issue by orlp** [pola-rs/polars#10300]
### Problem description

As brought up in https://github.com/pola-rs/polars/issues/10288, it is questionable whether the `return_dtype` should be optional on `apply` at all. It can lead to nasty bugs in production when all of a sudden a column is empty, and the `dtype` could not be determined.

**Comment by ritchie46** [comment]
Yes, I agree we should just require this. This can lead to subtle bugs and indeed doesn't fit our goal of a strict, explicit API.

**Comment by svaningelgem** [comment]
+1!

**Comment by coastalwhite** [comment]
I think this is stale now with #24027. `return_dtype` will now always be inferred.
